scripts/sample_model.py

```diff
@@ -0,0 +1,182 @@
+    pts_1 = (Point3D(0, 0, 3), Point3D(0, 10, 3), Point3D(10, 10, 3), Point3D(10, 0, 3))
+    pts_2 = (Point3D(10, 0, 3), Point3D(10, 10, 3), Point3D(20, 10, 3), Point3D(20, 0, 3))
+    pts_3 = (Point3D(0, 10, 3), Point3D(0, 20, 3), Point3D(10, 20, 3), Point3D(10, 10, 3))
+    pts_4 = (Point3D(10, 10, 3), Point3D(10, 20, 3), Point3D(20, 20, 3), Point3D(20, 10, 3))
+    room2d_1 = Room2D('Office 1', Face3D(pts_1), 3)
+    room2d_2 = Room2D('Office 2', Face3D(pts_2), 3)
+    room2d_3 = Room2D('Office 3', Face3D(pts_3), 3)
+    room2d_4 = Room2D('Office 4', Face3D(pts_4), 3)
+    story = Story('Office Floor', [room2d_1, room2d_2, room2d_3, room2d_4])
+    story.solve_room_2d_adjacency(0.01)
+    story.set_outdoor_window_parameters(SimpleWindowRatio(0.4))
+    story.properties.energy.construction_set = mass_set
+
+    dest_file = os.path.join(directory, 'story_simple.json')
+    with open(dest_file, 'w') as fp:
+        json.dump(story.to_dict(True), fp, indent=4)
+
+
+def room2d_simple(directory):
+    """Generate simple Room2D sample."""
+    mass_set = ConstructionSet('Thermal Mass Construction Set')
+    pts = (Point3D(0, 0, 3), Point3D(10, 0, 3), Point3D(10, 10, 3), Point3D(0, 10, 3))
+    ashrae_base = SimpleWindowRatio(0.4)
+    overhang = Overhang(1)
+    boundarycs = (bcs.outdoors, bcs.ground, bcs.outdoors, bcs.ground)
+    window = (ashrae_base, None, ashrae_base, None)
+    shading = (overhang, None, None, None)
+    room = Room2D('Shoe Box Zone', Face3D(pts), 3, boundarycs, window, shading)
+    room.properties.energy.construction_set = mass_set
+    room.properties.energy.program_type = office_program
+
+    dest_file = os.path.join(directory, 'room2d_simple.json')
+    with open(dest_file, 'w') as fp:
+        json.dump(room.to_dict(True), fp, indent=4)
+
+
+def context_shade_two_tree_canopy(directory):
+    """Generate the context_shade_two_tree_canopy sample."""
+    tree_canopy_geo1 = Face3D.from_regular_polygon(6, 6, Plane(o=Point3D(5, -10, 6)))
+    tree_canopy_geo2 = Face3D.from_regular_polygon(6, 2, Plane(o=Point3D(-5, -10, 3)))
+    tree_canopy = ContextShade('Tree Canopy', [tree_canopy_geo1, tree_canopy_geo2])
+
+    bright_leaves = ShadeConstruction('Bright Light Leaves', 0.5, 0.5, True)
+    tree_trans = ScheduleRuleset.from_constant_value(
+        'Tree Transmittance', 0.5, schedule_types.fractional)
+    
+    tree_canopy.properties.energy.construction = bright_leaves
+    tree_canopy.properties.energy.transmittance_schedule = tree_trans
+
+    dest_file = os.path.join(directory, 'context_shade_two_tree_canopy.json')
+    with open(dest_file, 'w') as fp:
+        json.dump(tree_canopy.to_dict(True), fp, indent=4)
+
+
+sample_directory = os.path.join('.', 'dragonfly_schema', 'samples')
+
+model_complete_simple(sample_directory)
+building_simple(sample_directory)
+story_simple(sample_directory)
+room2d_simple(sample_directory)
+context_shade_two_tree_canopy(sample_directory)
```

Comment:
Using `'.'` is tricky in this case. Since it will be different based on the folder that you call this module or in your case import it. You want to get the path of the current `__file__`.

```python
if __name__ == '__main__':

    sample_directory = os.path.join('.', 'dragonfly_schema', 'samples')

    model_complete_simple(sample_directory)
    building_simple(sample_directory)
    story_simple(sample_directory)
    room2d_simple(sample_directory)
    context_shade_two_tree_canopy(sample_directory)

Reply:
Are you saying that `export_samples.py` should be replaced with a batch or a shell file that runs each of the sample-generating Python files instead (I don't know how else I would get the ` if __name__ == '__main__'`)?  Or perhaps you meant to write:
```
__export_samples__ == '__main__'
```

Reply:
And I can replace the use of the . with the `__file__`.

Reply:
> Are you saying that export_samples.py should be replaced with a batch or a shell file that runs each of the sample-generating Python files instead

That's how you can run them from the command line. Generally speaking `if __name__ == '__main__' gives you the option to use a py file both by importing the module or running it as a script. My understanding is that the part for generating the sample files should run when you execute them from the command line and not when you import them as a module.

Reply:
Hm. I'm not sure that I like the batch file approach as much. A batch file will be harder to run on Travis if they are using Linux machines and, if we want to use .sh scripts, then generating the samples locally on our Windows computers, will require an extra step of installing bash. It just seems that it would be a lot nicer to have the automated sample file generation happen from a Python file and, if we go this route, treating the other Python files as modules seems to be the recommended approach:
https://stackoverflow.com/questions/7974849/how-can-i-make-one-python-file-run-another

In this case, I'm realizing the `__name__ == '__main__'` stuff definitely won't work. Do we really need the `__name__ == '__main__'` functionality here?  I guess another approach is just to write the `export_samples.py` so that it imports every single method from the other files and runs them.  It just means that we'll have to keep updating this export_samples file every time we add a new sample.

Reply:
I think my comment was confusing. I think it is fine either way. Let's just make sure to remove it from the package similar to tests by updating the setup.py file.

https://github.com/chriswmackey/dragonfly-schema/blob/master/setup.py#L20

Reply:
Ok. I also changed it to use the `__file__`  instead of `'.'` in case we ever want to run these methods from a file other than `export_samples`.

Reply:
Done!